presto-main/src/main/java/io/prestosql/sql/planner/LogicalPlanner.java

```diff
@@ -162,6 +163,11 @@ public Plan plan(Analysis analysis)
     }
 
     public Plan plan(Analysis analysis, Stage stage)
+    {
+        return plan(analysis, stage, analysis.getStatement() instanceof Explain || isCollectPlanStatisticsForAllQueries(session));
```

Comment:
Since `QueryExplainer` passes `true` here, do we need to have `instanceof`?
i don't like `instanceof`.

if this is about EXPLAIN ANALYZE, we should be able to pass this information in some better way.

Reply:
there is already similar check `io/prestosql/execution/SqlQueryExecution.java:410`

> if this is about EXPLAIN ANALYZE, we should be able to pass this information in some better way.

Maybe there is some other way, but is it worth it for such simple change?

Reply:
is this check here for EXPLAIN or EXPLAIN ANALYZE?

Reply:
> is this check here for EXPLAIN or EXPLAIN ANALYZE?

`EXPLAIN`, (any kind of).